intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java

```diff
@@ -0,0 +1,88 @@
+package tech.pantheon.yanginator.plugin.external;
+
+import static com.intellij.lang.parser.GeneratedParserUtilBase.enter_section_;
+import static com.intellij.lang.parser.GeneratedParserUtilBase.exit_section_;
+import static com.intellij.lang.parser.GeneratedParserUtilBase.recursion_guard_;
+
+import com.intellij.lang.PsiBuilder;
+import com.intellij.lang.parser.GeneratedParserUtilBase.Parser;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class ExternalRules {
+
+    /**
+     * First loop will go from first statement(word) to last, second loop will try to match every single item
+     * from listOfItems with the statement,
+     * if the item matches but offset does not change that means that the parsed word was optional. if we match optional word
+     * we save it to HashSet and try parsing it later. if it does match and offset moves that mean that it was parsed, so we
+     * can remove it.
+     * see more in /yanginator/plugin/external/AnyOrderRuleTutorial.md
+     *
+     * @param psiBuilder  Psi builder
+     * @param level       Level, by increasing it we move to the next element
+     * @param listOfItems list of items to try parsing and matching
+     * @return return true if all items are matched BUT does not check for duplicates
+     */
+
+    public static boolean anyOrder(PsiBuilder psiBuilder, int level, Parser... listOfItems) {
+        if (!recursion_guard_(psiBuilder, level, "rule")) return false;
+        boolean parserResultBoolean;
```

Comment:
Boolen in the name is not truly necessary as variable type is descriptive enough, better name would be "result"